### DISCUSSION TITLE
Sans I/O implementation of Websockets

### DISCUSSION THREAD

**Discussion by iudeen** [Kludex/uvicorn#1633] [Ideas] [answered]
I came across rather old discussion https://github.com/aaugustin/websockets/issues/975 about Uvicorn using legacy API of Websockets. 

Are we still interested in supporting the modern (better?) approach of implementing Websockets?

**Comment by Kludex** [comment, marked as answer]
We are, but you can implement it outside uvicorn, and if it gets attention, we can do a replacement at some point.

You can create your own WebSocket protocol, and tell uvicorn to use it: 
```python
import uvicorn

from ws_impl import IudeenWebSocketProtocol

if __name__ == "__main__":
    uvicorn.run("main.app", ws=IudeenWebSocketProtocol)
```

**Reply by iudeen** [reply]
Thanks for this!

This is really interesting!
